testing/hierarchies.py

```diff
@@ -588,6 +620,9 @@ def _visit_children(self, parent, child_spec):
                 child.def_in_file = parent
             else:
                 child.parent = parent
+                if child.kind == 'page' and parent.kind == 'page':
```

Comment:
A page needed both the parent to be defined as well as the `def_in_file` to get the tests to  work